common/djangoapps/static_replace/__init__.py

```diff
@@ -180,7 +180,7 @@ def replace_static_url(original, prefix, quote, rest):
             else:
                 # if not, then assume it's courseware specific content and then look in the
                 # Mongo-backed database
-                url = StaticContent.convert_legacy_static_url_with_course_id(rest, course_id)
+                url = StaticContent.get_canonicalized_asset_path(course_id, rest)
```

Comment:
It's not called from anywhere else, no.  I was really just thinking in the broader sense: URL generation/regeneration, other places where it might get used directly.  I guess I don't know the code well enough to really have a sense of how many other places we may or may not be canonicalizing asset paths.